cmd/storage-server/server.go

```diff
@@ -28,6 +28,7 @@ type Store interface {
 	Write(key string, val []byte) error
 	Has(key string) bool
 	Erase(key string) error
+	Keys(cancel <-chan struct{}) <-chan string
```

Comment:
I would change this API (down to the `diskv` implementation) to simply return a slice of strings, and not taking any cancellation struct.